logger.go

```diff
@@ -64,104 +61,121 @@ func (l *Logger) WithPrefix(prefix string) *Logger {
 func (l *Logger) Infof(format string, args ...interface{}) {
 	l.printf(infoLog, l.pfx(format), l.extend(args)...)
 }
 
-// InfofWithDepth is equivalent to the global InfofWithSepth function, with the addition of prefix and data content from this Logger.
-func (l *Logger) InfofWithDepth(extraDepth int, format string, args ...interface{}) {
-	l.printfWithDepth(infoLog, extraDepth, l.pfx(format), l.extend(args)...)
-}
-
 // Warning is equivalent to the global Warning function, with the addition of prefix and data content from this Logger.
 func (l *Logger) Warning(args ...interface{}) {
 	l.print(warningLog, l.extendWithPfx(args)...)
 }
 
-// WarningWithDepth is equivalent to the global WarningWithDepth function, with the addition of prefix and data content from this Logger.
-func (l *Logger) WarningWithDepth(extraDepth int, args ...interface{}) {
-	l.printWithDepth(warningLog, extraDepth, l.extendWithPfx(args)...)
-}
-
 // Warningln is equivalent to the global Warningln function, with the addition of prefix and data content from this Logger.
 func (l *Logger) Warningln(args ...interface{}) {
 	l.println(warningLog, l.extendWithPfx(args)...)
 }
 
-// WarninglnWithDepth is equivalent to the global WarninglnWithDepth function, with the addition of prefix and data content from this Logger.
-func (l *Logger) WarninglnWithDepth(extraDepth int, args ...interface{}) {
-	l.printlnWithDepth(warningLog, extraDepth, l.extendWithPfx(args)...)
-}
-
 // Warningf is equivalent to the global Warningf function, with the addition of prefix and data content from this Logger.
 func (l *Logger) Warningf(format string, args ...interface{}) {
 	l.printf(warningLog, l.pfx(format), l.extend(args)...)
 }
 
-// WarningfWithDepth is equivalent to the global WarningfWithDepth function, with the addition of prefix and data content from this Logger.
-func (l *Logger) WarningfWithDepth(extraDepth int, format string, args ...interface{}) {
-	l.printfWithDepth(warningLog, extraDepth, l.pfx(format), l.extend(args)...)
-}
-
 // Error is equivalent to the global Error function, with the addition of prefix and data content from this Logger.
 func (l *Logger) Error(args ...interface{}) {
 	l.print(errorLog, l.extendWithPfx(args)...)
 }
 
-// ErrorWithDepth is equivalent to the global ErrorWithDepth function, with the addition of prefix and data content from this Logger.
-func (l *Logger) ErrorWithDepth(extraDepth int, args ...interface{}) {
-	l.printWithDepth(errorLog, extraDepth, l.extendWithPfx(args)...)
+// GetErrorEvent is equivalent to the global GetErrorEvent function, with the addition of prefix and data content from this Logger.
+func (l *Logger) GetErrorEvent(args ...interface{}) Event {
+	return l.getEvent(errorLog, l.extendWithPfx(args))
+}
+
+// ErrorIf is equivalent to the global ErrorIf function, with the addition of prefix and data content from this Logger.
+func (l *Logger) ErrorIf(err error, args ...interface{}) {
+	if err != nil {
```

Comment:
glog.ErrorIfWithDepth(1, l.extendWithPfx(args)...) could be used here. In fact, all the methods could just be wrappers to glog.*WithDepth functions with the appropriate modifications to the args - less logic to worry about and less duplication. It may actually be the case that the embedded logger is not needed at all (that's just a speculation, but seems reasonable, as all the functionality was there without exposing it).